### DISCUSSION TITLE
Typing in song search automatically selects song

### DISCUSSION THREAD

**Discussion by jacques-jd** [ppy/osu#27667] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Is this intended behavior? In osu!stable, if you search in the search bar, it will update the list of beatmaps displayed to you but will NOT change the selected beatmap. This is useful behavior in case you're searching for something that doesn't exist and would like to go back to where you were before.

Another good use case of this is if you're applying filters to your beatmap selection. If I have a certain song picked, it will always get unpicked when I start typing key=7 since it'll start searching for the word "key" before the filter gets detected. 

### Version

2024.312.1-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14640371/compressed-logs.zip)

**Comment by peppy** [comment, marked as answer]
The way it's done right now is intentional, but I can understand why you'd prefer the stable behaviour.

We could either revert the behaviour if it's preferred by the majority, or alternatively add a setting. I think this kind of decision would come after we get around to updating song select with the newer design.

**Reply by jacques-jd** [reply]
Thanks for the feedback. I think a setting could prove beneficial.
